### DISCUSSION TITLE
groth16 verify not passed on zksync era

### DISCUSSION THREAD

**Discussion by yanyanho** [zkSync-Community-Hub/zksync-developers#422] [Compilers]
### Which compiler are you using?

- [X] zksolc
- [ ] zkvyper

### Which version of Solidity or Vyper?

solidity  0.8.20

### zksolc/zkvyper version?

latest

### Explain the issue

```bash
...
groth16 verify proof failed .  The failed transaction and the contract is verified . 

https://sepolia.explorer.zksync.io/tx/0x49cd81400530387f425fef00d73b20517eaaee2ad43eea875d9e9eb6462b4794

the contract code  is HappyRedPacket.sol . you can check here  : https://sepolia.explorer.zksync.io/address/0x195EA4d2218d37F15Aa7F24F910fDb64B5BA298C#contract

the groth16 verifier contract is here(auto-generated use snarkjs , it works on other layer2 Scroll , OP, Arbi) : https://sepolia.explorer.zksync.io/address/0x78bA4D3Ce180c585767Fa04A33D223280732e603#contract
```

### Link to your repo?

https://github.com/Dapp-Learning-DAO/Official-website-smart-contract/tree/main/zksync-redpacket

**Comment by 0xsisyfos** [comment]
Currently `ecPairing` is not supported. However, we are actively working on it, and the contract is being audited. Support for `ecPairing` should be available shortly, which should solve the issue for gross16 verify to pass.

**Reply by casiojapi** [reply]
hey, @0xsisyfos . Is this issue fixed? I'm having the same problem on testing a groth16 verifier on Sepolia.
